modules/nw-udn-cr.adoc

```diff
@@ -38,7 +38,7 @@ spec:
 <1> Name of your `UserDefinedNetwork` resource. This should not be `default` or duplicate any global namespaces created by the Cluster Network Operator (CNO).
 <2> The `topology` field describes the network configuration; accepted values are `Layer2` and `Layer3`. Specifying a `Layer2` topology type creates one logical switch that is shared by all nodes.
 <3> This field specifies the topology configuration. It can be `layer2` or `layer3`.
-<4> Specifies `Primary` or `Secondary`. `Primary` is the only `role` specification supported in 4.17.
```

Comment:
@JoeAldinger 

So based on
> The OVN-Kubernetes CNI plugin supports layer2, layer3, and localnet topologies for secondary pod networks. However, for the primary network, or the main network that all pods are attached to, only a layer3 topology is supported. This allows for network models where all pods in the cluster were part of the same global layer3 network, but restricts the ability to customize primary network configurations.

It sounds like `Primary` is invalid here, so we could omit this, or say that 

> That `Primary` is supported only for layer 2 topologies

Or some such, if I'm reading this right.

Also, that would mean the first example is invalid, because it is `layer2` and `Primary`, but the preceding text says that is a no-go. Would need QE to confirm this.

Reply:
@anuragthehatter can you advise on whether this example is valid. I'm pretty sure it is but want to check.

Reply:
@jab-rh I think you've got a good catch though in finding that the above text will also need to changed. Lets see what QE says. Thanks for the review.

Reply:
cc @Meina-rh who tested mix of such crd configs and also @ormergi to confirm this text. The example thought looks functional to me and QE has tested it

Reply:
If it does work, then we probably need to adjust the intro text that says that you cannot use `Primary` with layer 3 topologies, or maybe it does _work_, but we do not officially support it being used that way?

Reply:
Role Primary should be supported for both typologies (layer2 & layer3) in 4.17
Role Secondary should be supported in 4.18 (both layer2 & layer3).

Reply:
@ormergi and cc @Meina-rh does this mean that this is being back ported to 4.17? Meaning, should we have a PR in the works to land with a z-stream or is this supported right now and I should update this PR to reflect the current support? cc @jab-rh cc @stevsmit